xline/tests/lock_test.rs

```diff
@@ -0,0 +1,33 @@
+mod common;
+
+use std::{error::Error, time::Duration};
+
+use common::Cluster;
+use tokio::time;
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 10)]
+async fn test_lock() -> Result<(), Box<dyn Error>> {
```

Comment:
Please add a test to cover the deadlock case to ensure that the lease expired will release the lock